Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/AnimatedCursor.cs

```diff
@@ -59,7 +59,7 @@ public override void OnFocusChanged(FocusEventData eventData)
         {
             base.OnFocusChanged(eventData);
 
-            if (Pointer.CursorModifier != null)
+            if (Pointer?.CursorModifier != null)
```

Comment:
I believe many implementations of `IMixedRealityPointer` are MonoBehaviors, which would cause this check to behave potentially unexpectedly: https://blogs.unity3d.com/2014/05/16/custom-operator-should-we-keep-it/

Reply:
I agree that it is problematic - is there an existing MRTK convention for handling this case? We have a have a pretty simple NullIfDestroyed extension in our codebase for this situation (it just tries casting to a MonoBehaviour or Unity.Object and then does the overloaded null check on the casted version), and I could bring that into MRTK if it would help.

Reply:
in general, we have used a two step validation

if (Pointer != null && Pointer.CursorModifier != null)

Reply:
That works if you're doing the null check through a class type, but not through an interface type. The interface ==/!= is still just doing a reference (in)equality check, without the extra "is destroyed" check that the Unity.Object ==/!= null will do (https://stackoverflow.com/questions/1881459/how-do-i-override-the-equals-operator-for-an-interface-in-c).